include/openthread/srp_client.h

```diff
@@ -245,6 +260,60 @@ const otSockAddr *otSrpClientGetServerAddress(otInstance *aInstance);
  */
 void otSrpClientSetCallback(otInstance *aInstance, otSrpClientCallback aCallback, void *aContext);
 
+/**
+ * This function enables the auto-start mode.
+ *
+ * This is only available when auto-start feature `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE` is enabled.
+ *
+ * Config option `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_MODE` specifies the default auto-start mode (whether
+ * it is enabled or disabled at the start of OT stack).
+ *
+ * When auto-start is enabled, the SRP client will monitor the Thread Network Data for SRP Server Service entries
+ * and automatically start and stop the client when an SRP server is detected.
+ *
+ * If multiple SRP servers are found, a random one will be selected. If the selected SRP server is no longer
```

Comment:
This is still TBD, but if there are multiple SRP servers, it may be necessary to register with all - for example, in multi-homed environments.

Reply:
The current SRP client implementation supports registering items (service/host info) with a single server.

We can consider extending it to register with multiple servers in parallel but we may then need to re-think some aspect of its design (and its APIs). For example, we probably need to maintain a separate state per service/host item for each SRP server.